src/solver/simulation/solver.hxx

```diff
@@ -717,9 +715,10 @@ void ISimulation<Impl>::computeRandomNumbers(randomNumbers& randomForYears,
             // Previous month's first day in the year
             int firstDayOfMonth = study.calendar.months[initResLevelOnSimMonth].daysYear.first;
 
-            double randomLevel = hydroManagement.randomReservoirLevel(min[firstDayOfMonth],
+            double randomLevel = HydroManagement::randomReservoirLevel(min[firstDayOfMonth],
```

Comment:
Can we make it a free function instead of a static function ?
Since it has nothing to do anymore with the `HydroManagement` class, it will make things clearer.